interface/main/tabs/main.php

```diff
@@ -86,6 +86,9 @@
         // used in tabs_view_model.js.
         jsGlobals.enable_group_therapy = <?php echo js_escape($GLOBALS['enable_group_therapy']); ?>
 
+        var WindowTitleAddPatient = <?php echo ($GLOBALS['window_title_add_patient_name'] ? 'true' : 'false' ); ?>;
+        var WindowTitleBase = "<?php echo text($openemr_name); ?>";
```

Comment:
You would want to use js_escape here instead of text.  There are different security vulnerabilities in javascript versus when we are escaping in html.

Reply:
does js_escape escape proper names properly? i.e hyphens, accents and other characters that get thrown at use from different languages/countries.
I'm unsure but it seems I've had this issue in the past.  Also, js_escape will double quote.

Reply:
should look like this (js_escape without the outer double quotes):
```
var WindowTitleBase = <?php echo js_escape($openemr_name); ?>;
```
(if find any use cases where this is breaking down, definitely let me know; at this point i don't know of it breaking anything; it does require utf8 encoding (specifically for when using other characters), which is now pretty much only encoding used in browsers anyways)

Reply:
@bradymiller

Using js_escape breaks the code.

There are several places where $openemr_name is echo'd and none of them use js_escape.  And there is a reason for that.

Look at line 53 of main.php.  js_escape is not used there.  text()  is used there.

The function text() escapes the argument by calling htmlspecialchars.

echo js_escape($openemr_name) gives a result with double quotes.  The result of the echo is ""OpenEMR"", resulting in an error.

PS I did update the function clearPatient() and that change works.  Thanks for pointing that out.

Reply:
hi @psoas , nice feature and work on this.
I checked out your code and made the suggested change to 
```
var WindowTitleBase = <?php echo js_escape($openemr_name); ?>;
```
and it's working fine.

The line in main.php is in html context vs this line which is in javascript.

Reply:
oops, I did not realize that I had the quote marks outside of <?php echo js_escape($openemr_name); ?>.  When I saw the double quotes, I should have noticed that and deleted those.

Reply:
Hi @sjpadgett @stephenwaite and @bradymiller

Should it be:
echo js_escape(text($openemr_name));

Or, is that redundant?

Reply:
Yep that'd be double escaping. Just js_escape should be fine.